lib/cylc/task_events_mgr.py

```diff
@@ -524,8 +524,7 @@ def _event_email_callback(self, proc_ctx, schd_ctx):
                 else:
                     self.event_timers[id_key].unset_waiting()
             except KeyError as exc:
-                if cylc.flags.debug:
-                    LOG.exception(exc)
+                LOG.exception(exc)
```

Comment:
`KeyError` used to happen quite often because we were relying on the task proxy being in the pool. In the current logic, this is no longer the case - as the event handler contexts are now kept in a dict independent of their original task proxies.

Reply:
So if an exception is caught, we want to log it, regardless.

Reply:
Thanks for the info!